elastic/kibana
# [Infra] Change the inventory ml license upgrade prompt

Closes #176502 
## Summary

This PR changes the inventory license upgrade prompt. Now the prompt will use the `licenseManagement` plugin to get the locator and navigate to the license management page (using the 'Start trial' button) - a fallback path is provided as `licenseManagement` plugin is optional. With this change the locator will be used also for the logs UI prompt (which is still using the old prompt design).
To avoid complicating the logic and infra and improve the consistency we will now have a single prompt with the same content (we won't do the trial check and will let the license management page decide what options to show after navigating)

| Before | After |
| ------ | ------ |
| <img width="809" alt="image" src="https://github.com/elastic/kibana/assets/14139027/275f4a8c-cd9f-4e85-a43c-9beff4b69ccb"> | <img width="1920" alt="image" src="https://github.com/elastic/kibana/assets/14139027/9ed0cd8a-0f98-4f2f-af95-45fe6ea181b9"> |

## Testing 
- Run Kibana with a basic license
   - Use `POST /_license/start_basic?acknowledge=true` to set basic license
- Go to Infrastructure > Inventory and click on `Anomalies Detection`
   - Click on the Start Trial and check the page (The action there will be different: Update/Start trial based on the user actions)

https://github.com/elastic/kibana/assets/14139027/a7dacd19-d29f-4cf9-b5d8-ff1b9a6e8440
